server/cmwell-data-tools/src/main/scala/cmwell/tools/data/utils/akka/Retry.scala

```diff
@@ -113,21 +124,45 @@ object Retry extends DataToolsLogging with DataToolsConfig {
             e.discardBytes()
             None
           } else {
-            e.discardBytes()
+            count match {
+              case Some(c) if c > 0 =>
 
-            // schedule a retry to http stream
-            logger.warn(
-              s"$labelValue server error: will retry again in $delay host=${getHostname(h)} status=$s entity=$e"
-            )
-            val future = after(delay, system.scheduler)(Future.successful(data))
-            Some(immutable.Seq(future -> state))
+                val retryBackoff = delayWithFactor(delayFactor,c,iterationDelay,limit.get)
+
+                e.discardBytes()
```

Comment:
Requested by @e-orz : we need to expose response body in log. We can do the trick with a "log ID" that will be printed now, and the rest once the body of the response is consumed.

Reply:
Please can we discuss this some more..